subsys/net/lib/nrf_cloud/src/nrf_cloud_fsm.c

```diff
@@ -491,8 +499,24 @@ static int dc_rx_data_handler(const struct nct_evt *nct_evt)
 		.data = nct_evt->param.dc->data,
 		.topic = nct_evt->param.dc->topic,
 	};
+	/* All data is forwared to the app... unless a callback is registered */
+#if defined(CONFIG_NRF_CLOUD_CELL_POS) && defined(CONFIG_NRF_CLOUD_MQTT)
+	if (cell_pos_cb) {
+		struct nrf_cloud_cell_pos_result res;
+		int ret = nrf_cloud_cell_pos_process(nct_evt->param.dc->data.ptr, &res);
+
+		if (ret <= 0) {
+			if (ret == 0) {
+				LOG_DBG("Passing cell pos data to cb");
+				/* Successfully parsed, send to callback */
+				cell_pos_cb(&res);
+			}
+			return 0;
+		}
```

Comment:
I was concerned about this, because it doesn't send the received data to the application if `nrf_cloud_cell_pos_process()` returns a negative value, but looking at the code, I think that should be fine. The function should return 1 unless the data contains a valid cellular position or the data looks like cellular position but is malformed.